src/hssm/plotting/posterior_predictive.py

```diff
@@ -0,0 +1,199 @@
+"""Plotting functionalities for HSSM."""
+
+import logging
+
+import arviz as az
+import matplotlib as mpl
+
+# import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import seaborn as sns
+
+from .utils import _get_plotting_df
+
+_logger = logging.getLogger("hssm")
+
+
+def plot_posterior_predictive(
+    model,
+    idata: az.InferenceData | None = None,
+    data: pd.DataFrame | None = None,
+    n_samples: int | float | None = 20,
+    row: str | None = None,
+    col: str | None = None,
+    col_wrap: int | None = None,
+    palette: str | list | dict | None = None,
+    binwidth: float | tuple[float, float] | None = 0.1,
+    grid_kwargs: dict | None = None,
+    title: str | None = None,
+    xlabel: str | None = "Response Time",
+    ylabel: str | None = None,
+    **kwargs,
+) -> mpl.axes.Axes | sns.FacetGrid:
+    """Plot the posterior predictive distribution against the observed data.
+
+    Parameters
+    ----------
+    model : hssm.HSSM
+        A fitted HSSM model.
+    idata : optional
+        The InferenceData object with posterior samples. If not provided, will use the
+        traces object stored inside the model. If posterior predictive samples are not
+        present in this object, will generate posterior predictive samples using the
+        this InferenceData object and the original data.
+    data : optional
```

Comment:
Data here has two functions:

1. Specify the covariates to allow posterior predictive sampling (potentially out-of-sample --> pure predictions) (note that we can literally make the values of the covariates up, and still get posterior predictive samples from our model)
2. Supply the real `rts, responses`. This is used only when we want to add real data to our plot.

Note that the first aspect can be used to generate posterior predictive samples WITHOUT having any real `rts, responses`.

The second aspect should be optional in this plot. 
We should be able to get the posterior predictive plot with and without overlay of real `rts, responses`.